### ISSUE TITLE
Flash mass balance violated by independent x,y normalization

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/differentiable-flowsheets#80]
## Description

In src/difflow/units/flash.py, the liquid and vapor mole fractions (x_i, y_i) are normalized independently after the flash calculation. This can violate the overall component mass balance:

z_i * F = x_i * L + y_i * V

If x and y are normalized independently (dividing each by their respective sums), the component balance no longer closes exactly. The error is typically small but can accumulate in multi-stage calculations (distillation).

## Impact

Small mass balance errors in flash calculations that propagate through multi-stage distillation columns.

## Suggested Fix

Use the Rachford-Rice equation to solve for vapor fraction, which inherently satisfies both the component balance and the sum-of-fractions constraints simultaneously.

## Severity
MODERATE

## Component
Core - Separation (src/difflow/units/flash.py)

**Comment by jkitchin** [comment]
## Fixed in commit 0f395a1

**Root cause:** After the Rachford-Rice solve, both liquid (x) and vapor (y) compositions were independently normalized:
```python
x = x / jnp.sum(x)
y = y / jnp.sum(y)
```
Independent normalization breaks the component mass balance `z_i * F = x_i * L + y_i * V` because the normalization factors for x and y are generally different.

**Fix:** Normalize only x, then derive y from the component mass balance:
```python
x = x / jnp.sum(x)
y = (z - x * (1 - V_frac)) / V_frac
```

This ensures **exact** mass balance closure: `z_i = x_i * (1 - V) + y_i * V` for all components.

**Example:**
```python
# Binary system: z = [0.5, 0.5], V/F = 0.4
# Old: x = [0.62, 0.38], y = [0.33, 0.67]
#   Check: 0.62*0.6 + 0.33*0.4 = 0.504 ≠ 0.5  (mass balance violated!)
# New: x = [0.62, 0.38], y derived from balance
#   Check: z_i = x_i*(1-V) + y_i*V exactly  ✓
```

**Test:** `tests/test_core_unit_bugs.py`
